src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -513,83 +514,78 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 						overrideResult.editor,
 						overrideResult.options ?? resolvedOptions
 					);
 				}
 			}
 
+			const resolvedEditor = editor instanceof EditorInput ? editor : this.createEditorInput(editor);
 			// Override handling: disabled or no override found
 			return resolvedGroup.openEditor(resolvedEditor, resolvedOptions);
 		}
 
 		return undefined;
 	}
 
-	private doResolveEditorOpenRequest(editor: IEditorInput | IUntypedEditorInput, optionsOrGroup?: IEditorOptions | OpenInEditorGroup, group?: OpenInEditorGroup): [IEditorGroup, EditorInput, IEditorOptions | undefined] | undefined {
+	private doResolveEditorOpenRequest(editor: IEditorInput | IUntypedEditorInput, optionsOrGroup?: IEditorOptions | OpenInEditorGroup, group?: OpenInEditorGroup): [IEditorGroup, IEditorOptions | undefined] | undefined {
 		let resolvedGroup: IEditorGroup | undefined;
 		let candidateGroup: OpenInEditorGroup | undefined;
 
-		let typedEditor: EditorInput | undefined;
 		let options: IEditorOptions | undefined;
 
 		// Typed Editor Support
 		if (editor instanceof EditorInput) {
-			typedEditor = editor;
 			options = optionsOrGroup as IEditorOptions;
 
 			candidateGroup = group;
-			resolvedGroup = this.findTargetGroup(typedEditor, options, candidateGroup);
+			resolvedGroup = this.findTargetGroup(editor, options, candidateGroup);
 		}
 
 		// Untyped Text Editor Support
 		else {
 			const textInput = editor as IUntypedEditorInput;
-			typedEditor = this.createEditorInput(textInput);
-			if (typedEditor) {
-				options = textInput.options;
+			candidateGroup = optionsOrGroup as OpenInEditorGroup;
+			options = textInput.options;
+			resolvedGroup = this.findTargetGroup(textInput, options, candidateGroup);
+		}
 
-				candidateGroup = optionsOrGroup as OpenInEditorGroup;
-				resolvedGroup = this.findTargetGroup(typedEditor, options, candidateGroup);
+		if (
+			this.editorGroupService.activeGroup !== resolvedGroup && 	// only if target group is not already active
+			options && !options.inactive &&								// never for inactive editors
+			options.preserveFocus &&									// only if preserveFocus
+			typeof options.activation !== 'number' &&					// only if activation is not already defined (either true or false)
+			candidateGroup !== SIDE_GROUP								// never for the SIDE_GROUP
+		) {
+			// If the resolved group is not the active one, we typically
+			// want the group to become active. There are a few cases
+			// where we stay away from encorcing this, e.g. if the caller
+			// is already providing `activation`.
+			//
+			// Specifically for historic reasons we do not activate a
+			// group is it is opened as `SIDE_GROUP` with `preserveFocus:true`.
+			// repeated Alt-clicking of files in the explorer always open
+			// into the same side group and not cause a group to be created each time.
+			options.activation = EditorActivation.ACTIVATE;
+
+			if (!(editor instanceof EditorInput)) {
+				(editor as IUntypedEditorInput).options = options;
```

Comment:
Isn't this replacing all of the `options` that were maybe passed in? Why not mixing them in?

Reply:
Oh wow, this is confusing, the `options` are from the untyped input. Then the code is not doing anything right?

Reply:
The reason I do this is just to get the added activation property.

Reply:
This no longer applies.